lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -151,15 +151,48 @@ def sniff!
     ES2_SNIFF_RE_URL  = /([^\/]*)?\/?([^:]*):([0-9]+)/
     # Sniffs and returns the results. Does not update internal URLs!
     def check_sniff
-      url, resp = perform_request(:get, '_nodes')
+      url, resp = perform_request(:get, '_nodes/http')
```

Comment:
This looks like a behavioral change (selecting only http nodes)?

Also, I can't find docs on the /_nodes/http endpoint. I think I found where it's defined in the Elasticsearch source code, but not in the docs.